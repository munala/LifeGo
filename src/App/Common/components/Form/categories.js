export default [
  'Art & Design',
  'Auto & Vehicles',
  'Books',
  'Business',
  'Dating',
  'Education',
  'Entertainment',
  'Events',
  'Finance',
  'Food & Drink',
  'Health & Fitness',
  'House & Home',
  'Music',
  'Photography',
  'Shopping',
  'Social',
  'Sports',
  'Travel & Local',
  'Adventure',
  'Other',
];
