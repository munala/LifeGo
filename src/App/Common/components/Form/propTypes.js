import PropTypes from 'prop-types';

export default {
  bucketlist: PropTypes.shape({
    name: PropTypes.string,
    pictureUrl: PropTypes.string,
  }),
  onClose: PropTypes.func.isRequired,
  open: PropTypes.bool.isRequired,
  save: PropTypes.func.isRequired,
  profile: PropTypes.shape({}).isRequired,
};
