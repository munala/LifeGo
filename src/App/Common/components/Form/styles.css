.container {
  display: flex;
  justify-content: center;
  align-items: center;
}
.form-container{
  overflow: scroll;
  display: flex;
  flex-direction: column;
  margin: auto;
  justify-content: space-between;
  align-items: stretch;
  background-color: #fff;
  padding-top: 0px;
  border-radius: 10px;
  min-width: 500px;
}

.input-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: stretch;
  padding-left: 10px;
  padding-right: 10px;
}

.form-title {
  display: flex;
  color:#fff;
  text-align: center;
  padding-left: 20px;
  vertical-align: middle;
  height: 50px;
  line-height: 50px;
  background-color: #00bcd4;
  box-shadow: 0 0 2px 2px rgba(0,0,0,0.4);
  margin-bottom: 5px;
}

.short-inputs{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items:baseline;
}

.actions{
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  background-color: #f7f7f7;
}

.inputfile {
	width: 0.1px;
	height: 0.1px;
	opacity: 0;
	overflow: hidden;
	position: absolute;
	z-index: -1;
}
.inputfile + label {
    font-size: 1.25em;
    font-weight: 700;
    color: white;
    background-color: black;
    display: inline-block;
}

.inputfile:focus + label,
.inputfile + label:hover {
    background-color: red;
}
.inputfile + label {
	cursor: pointer;
}

.photo-field {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  margin-left: 10px;
  margin-right: 10px;
  margin-bottom: 10px;
  font-size: 1rem;
}
.photo-field-label {
  display: flex;
  color: #757575;
  margin-bottom: 5px
}

.photo-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: stretch;
  border-radius: 10px;
  overflow: hidden;
  background-color: #f7f7f7;
  height: 200px;
  width: 200px;
}

.photo {
  width: 100%;
  height: 100%;
  object-fit:cover;
  opacity: 0.8;
}

.photo:hover {
  opacity: 1;
}

.photo-actions {
  width: 200px;
  position: absolute;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  background-color: rgba(0,0,0,0.5);
  border-radius: 0 0 10px 10px
}

.photo-actions:hover {
  background-color: rgba(0,0,0,0.7);
}

.photo-input-label {
  display: flex;
  color: #fff;
  padding: 10px;
  text-align: center;
  cursor: pointer;
  text-decoration: none;
}

.location {
  display: flex;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 20px;
  padding-left: 10px;
  padding-right: 10px;
}

.location-icon {
  margin-right: 8px;
}

.fa.fa-map-marker.header {
  color: #e53935;
}

.location-input{
  width: 100%;
  border-width: 0px;
  font-size: 1rem;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  background-color: transparent;
  display: flex;
  padding-bottom: 6px;
  border-bottom: 1px solid #949494;
}

.location-input:focus{
  padding-bottom: 5px;
  border-bottom: 2px solid rgb(0, 131, 148);
}
.location-input:hover{
  padding-bottom: 5px;
  border-bottom: 2px solid rgb(0.0, 0, 0.2);
}
.location-input:active,
 {
   padding-bottom: 5px;
  border-bottom: 2px solid rgb(0, 131, 148);
}

.location-input::placeholder {
  color: #A2A2A2;
  font-size: 1rem;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-weight:300;
}

.location-drop-down {
  cursor: pointer;
  width: 100%;
  border-width: 0px;
  z-index: 100;
}

.suggestion {
  font-size: 1rem;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  padding: 4px;
  text-align: left;
  background-color: #f7f7f7;
  cursor: pointer;
}

.suggestion:hover {
  background-color: #eee;
  cursor: pointer;
}

.location-footer {
  display: flex;
  justify-content: flex-end;
  padding: 4px;
}

.location-footer-image {
  display: inline-block;
  width: 150px;
}

@media (max-width: 480px) {
  .autocomplete-container {
    text-align: left;
    display: flex;
  }
  .form-container {
    min-width: 200px;
  }
}

@media (min-width: 768px) {
  .alert {
    margin-left: auto;
    margin-right: auto;
    width: 50%;
    -webkit-animation: fadein .75s;
    -moz-animation: fadein .75s;
    -ms-animation: fadein .75s;
    -o-animation: fadein .75s;
    animation: fadein .75s;
  }

  .btn {
    margin-top: 1.5rem;
  }

  .location {
    display: inline-block;
    margin-left: 0px;
    margin-right: 0px;
    margin-top: 20px;
  }
  .form-container {
    min-width: 500px;
  }
}

.geocoding-results {
  text-align: left;
}

@keyframes fadein {
  from { opacity: 0; }
  to   { opacity: 1; }
}

@-moz-keyframes fadein {
  from { opacity: 0; }
  to   { opacity: 1; }
}

@-webkit-keyframes fadein {
  from { opacity: 0; }
  to   { opacity: 1; }
}

@-ms-keyframes fadein {
  from { opacity: 0; }
  to   { opacity: 1; }
}

@-o-keyframes fadein {
  from { opacity: 0; }
  to   { opacity: 1; }
}

::placeholder { /* Chrome, Firefox, Opera, Safari 10.1+ */
    color: #ccc;
    opacity: 1; /* Firefox */
}

:-ms-input-placeholder { /* Internet Explorer 10-11 */
    color: #ccc;
}

::-ms-input-placeholder { /* Microsoft Edge */
    color: #ccc;
}

input:focus, textarea:focus, select:focus, button:focus{
    outline: none;
}

button {
  border: none;
  background-color: transparent;
}

.link {
  text-decoration: none;
  color: #00bcd4;
}

@-webkit-keyframes grow {
    from {width: 0%;}
    to {width: 100%;}
}

@keyframes grow {
  from {width: 0%;}
  to {width: 100%;}
}

@-webkit-keyframes card-grow {
    from {height: 0vh;}
    to {height: 100%;}
}

@keyframes card-grow {
    from {height: 0vh;}
    to {height: 100%;}
}

@-webkit-keyframes profile-loading {
    from {background-color: #f7f7f7;}
    to {background-color: #ffffff;}
}

@keyframes profile-loading {
    from {background-color: #f7f7f7;}
    to {background-color: #ffffff;}
}
