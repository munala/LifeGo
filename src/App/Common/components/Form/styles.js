export default {
  input: {
    display: 'flex',
    justifyContent: 'center',
    alignSelf: 'center',
    width: 'calc(100% - 40px)',
  },
  formModal: {
    height: '100%',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    display: 'flex',
  },
};
