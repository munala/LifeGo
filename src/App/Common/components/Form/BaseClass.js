/* eslint-disable react/no-unused-state */
import { Component } from 'react';

import propTypes from './propTypes';

class BaseClass extends Component {
  onChange = ({ value, type }) => {
    const bucketlist = { ...this.state.bucketlist };
    bucketlist[type] = value || null;

    const disabled = !bucketlist.name && bucketlist[type] === this.props.bucketlist[type];

    this.setState({
      bucketlist,
      disabled,
    });
  }

  onSave = async () => {
    const { image, bucketlist } = this.state;
    if (image) {
      const { url: pictureUrl } = await this.uploadImage(image);
      bucketlist.pictureUrl = pictureUrl;
    }

    this.props.save(bucketlist);
    this.setState({ bucketlist: initialBucketlist });
  }

  changePhoto = ({ target: { files } }) => {
    if (files && files.length > 0) {
      const [image] = files;
      if (image) {
        const reader = new FileReader();

        reader.readAsDataURL(image);

        reader.onloadend = (e) => {
          this.setState({
            imageUrl: reader.result,
          });
        };

        const { bucketlist } = this.state;

        this.setState({
          image,
          disabled: !bucketlist.name,
        });
      }
    }
  }

  removePhoto = () => {
    this.setState({
      image: null,
      imageUrl: '',
    });
  }

  uploadImage = async image => new Promise((resolve, reject) => {
    const apiKey = process.env.REACT_APP_CLOUDINARY_KEY;
    const formData = new FormData();

    formData.append('file', image);
    formData.append('upload_preset', 'dl5sqcqz');
    formData.append('api_key', apiKey);
    formData.append('mode', 'no-cors');

    let response;
    const xhr = new XMLHttpRequest();
    xhr.open('POST', 'https://api.cloudinary.com/v1_1/lifego/image/upload', true);
    xhr.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
    xhr.send(formData);
    xhr.onreadystatechange = (e) => {
      if (xhr.readyState === 4 && xhr.status === 200) {
        response = JSON.parse(xhr.responseText);
        resolve(response);
      }
    };
  });
}

BaseClass.propTypes = propTypes;

export const initialBucketlist = {
  name: '',
  description: '',
  category: '',
  dueDate: '',
  location: '',
  privacy: 'everyone',
};

export default BaseClass;
