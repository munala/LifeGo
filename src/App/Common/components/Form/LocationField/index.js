import React from 'react';
import PropTypes from 'prop-types';
import PlacesAutocomplete from 'react-places-autocomplete';

import { renderSuggestion } from './autoCompleteProps';
import '../styles.css';

const LocationField = ({
  bucketlist,
  onChange,
}) => (
  <PlacesAutocomplete
    value={bucketlist.location || ''}
    onChange={value => onChange({ value, type: 'location' })}
    onSelect={value => onChange({ value, type: 'location' })}
  >
    {({ getInputProps, suggestions, getSuggestionItemProps }) => (
      <div className="location-container">
        <input
          {...getInputProps({
            placeholder: 'type location',
            className: 'location-input',
          })}
        />
        <div className="location-drop-down">
          {suggestions.map(suggestion => renderSuggestion(suggestion, getSuggestionItemProps))}
        </div>
      </div>
    )}
  </PlacesAutocomplete>
);
LocationField.propTypes = {
  bucketlist: PropTypes.shape({}).isRequired,
  onChange: PropTypes.func.isRequired,
};

export default LocationField;
