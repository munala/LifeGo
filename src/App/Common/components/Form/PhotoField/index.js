/* eslint-disable jsx-a11y/label-has-for */
import React from 'react';
import PropTypes from 'prop-types';
import '../styles.css';

const PhotoField = ({
  bucketlist,
  changePhoto,
  removePhoto,
  imageUrl,
}) => (
  <div className="photo-field">
    <div className="photo-field-label">photo</div>
    <div className="photo-wrapper" onClick={changePhoto}>
      {
        imageUrl &&
        <img
          id="target"
          className="photo"
          alt={bucketlist.name}
          src={imageUrl}
        />
      }
      <div className="photo-actions">
        <label htmlFor="file-upload" className="photo-input-label">
          {imageUrl ? 'change' : 'add'}
        </label>
        <input id="file-upload" className="inputfile" type="file" onChange={changePhoto} />
        {imageUrl && <a href="#" className="photo-input-label" onClick={removePhoto}>remove</a>}
      </div>
    </div>
  </div>
);
PhotoField.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  bucketlist: PropTypes.shape({}).isRequired,
  changePhoto: PropTypes.func.isRequired,
  removePhoto: PropTypes.func.isRequired,
};

export default PhotoField;
