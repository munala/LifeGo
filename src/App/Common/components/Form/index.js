import React from 'react';
import Modal from '@material-ui/core/Modal';
import MenuItem from '@material-ui/core/MenuItem';

import categories from './categories';
import TextInput from '../TextInput';
import FlatButton from '../FlatButton';
import DatePicker from '../DatePicker';
import PhotoField from './PhotoField';
import LocationField from './LocationField';
import BaseClass, { initialBucketlist } from './BaseClass';
import propTypes from './propTypes';
import './styles.css';

class Form extends BaseClass {
  state = {
    disabled: true,
    bucketlist: this.props.bucketlist,
    saving: false,
    error: false,
    image: null,
    imageUrl: this.props.bucketlist.pictureUrl || '',
  };

  render() {
    const { onClose, open } = this.props;

    const {
      bucketlist, error, disabled, saving, imageUrl,
    } = this.state;

    return (
      <Modal
        open={open}
        onClose={onClose}
      >
        <div className="form-container">
          <div className="form-title">{`${this.props.bucketlist.name ? 'Edit' : 'Add'} bucketlist`}</div>
          <div className="input-wrapper">
            <TextInput
              autoFocus
              name="name"
              placeholder="name of your bucketlist"
              value={bucketlist.name}
              onChange={({ target: { value } }) => this.onChange({ value, type: 'name' })}
              error={error}
            />
            <TextInput
              name="description"
              placeholder="tell people more about your bucketlist"
              value={bucketlist.description}
              onChange={({ target: { value } }) => this.onChange({ value, type: 'description' })}
              multiline
              rows={4}
              rowsMax={4}
            />
            <DatePicker
              label="due date"
              onChange={({ target: { value } }) => this.onChange({ value, type: 'dueDate' })}
              defaultValue={bucketlist.dueDate}
            />
            <TextInput
              name="category"
              value={bucketlist.category}
              select
              label="select category"
              onChange={({ target: { value } }) => this.onChange({ value, type: 'category' })}
            >
              {categories.map(category => (
                <MenuItem key={category} value={category}>
                  {category}
                </MenuItem>
              ))}
            </TextInput>
            <LocationField
              bucketlist={bucketlist}
              onChange={this.onChange}
            />
            <PhotoField
              bucketlist={bucketlist}
              changePhoto={this.changePhoto}
              imageUrl={imageUrl}
              removePhoto={this.removePhoto}
            />
          </div>
          <div className="actions">
            <FlatButton
              label="cancel"
              style={{ color: 'grey' }}
              onClick={onClose}
            />
            <FlatButton
              label="add"
              keyboardFocused
              onClick={this.onSave}
              disabled={saving || disabled}
            />
          </div>
        </div>
      </Modal>
    );
  }
}

Form.propTypes = propTypes;

Form.defaultProps = {
  bucketlist: initialBucketlist,
};

export default Form;
