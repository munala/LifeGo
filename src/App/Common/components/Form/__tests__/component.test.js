import React from 'react';
import { shallow } from 'enzyme';
import axios from 'axios';

import Form from '../';

jest.mock('axios');
axios.post = jest.fn().mockReturnValue({ url: 'image' });

describe('Form tests', () => {
  let wrapper;
  const onClose = jest.fn();
  const save = jest.fn();
  const bucketlist = {
    name: '',
    description: '',
    category: '',
    dueDate: '',
    location: '',
  };

  beforeAll(() => {
    wrapper = shallow(<Form
      onClose={onClose}
      open
      save={save}
    />);
  });

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('handles text change', () => {
    const inputs = wrapper.find('WithStyles(TextInput)');
    inputs.first().props().onChange({ target: { value: 'value' } });
    expect(wrapper.instance().state.bucketlist.name).toEqual('value');
  });

  it('handles submitting', () => {
    const inputs = wrapper.find('WithStyles(TextInput)');
    inputs.first().props().onChange({ target: { value: 'value' } });

    const buttons = wrapper.find('WithStyles(FlatButton)');
    buttons.last().props().onClick();

    expect(save).toHaveBeenCalledTimes(1);
    expect(save).toHaveBeenCalledWith({ ...bucketlist, name: 'value' });
  });

  it('handles  photos', () => {
    const inputs = wrapper.find('WithStyles(TextInput)');
    inputs.first().props().onChange({ target: { value: 'value' } });
    const photoField = wrapper.find('PhotoField');

    const fileContents = 'name: image';
    const image = new Blob([fileContents], { type: 'image' });

    // adding photo
    photoField.props().changePhoto({ target: { files: [image] } });
    expect(wrapper.instance().state.image).toEqual(image);

    // uploading photo
    const buttons = wrapper.find('WithStyles(FlatButton)');
    buttons.last().props().onClick();
    expect(axios.post).toHaveBeenCalledTimes(1);

    // remove photo
    photoField.props().removePhoto();
    expect(wrapper.instance().state.image).toEqual(null);
  });
});
