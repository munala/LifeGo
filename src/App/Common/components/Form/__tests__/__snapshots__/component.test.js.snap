// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form tests renders without crashing 1`] = `
<WithStyles(Modal)
  onClose={[Function]}
  open={true}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <div
    className="form-container"
  >
    <div
      className="form-title"
    >
      Add bucketlist
    </div>
    <div
      className="input-wrapper"
    >
      <TextField
        autoFocus={true}
        error={false}
        name="name"
        onChange={[Function]}
        placeholder="name of your bucketlist"
        required={false}
        select={false}
        style={
          Object {
            "alignSelf": "center",
            "display": "flex",
            "justifyContent": "center",
            "width": "calc(100% - 40px)",
          }
        }
        value=""
      />
      <TextField
        multiline={true}
        name="description"
        onChange={[Function]}
        placeholder="tell people more about your bucketlist"
        required={false}
        rows={4}
        rowsMax={4}
        select={false}
        style={
          Object {
            "alignSelf": "center",
            "display": "flex",
            "justifyContent": "center",
            "width": "calc(100% - 40px)",
          }
        }
        value=""
      />
      <WithStyles(DatePicker)
        defaultValue=""
        label="due date"
        onChange={[Function]}
      />
      <TextField
        label="select category"
        name="category"
        onChange={[Function]}
        required={false}
        select={true}
        style={
          Object {
            "alignSelf": "center",
            "display": "flex",
            "justifyContent": "center",
            "width": "calc(100% - 40px)",
          }
        }
        value=""
      >
        <WithStyles(MenuItem)
          key="Art & Design"
          value="Art & Design"
        >
          Art & Design
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Auto & Vehicles"
          value="Auto & Vehicles"
        >
          Auto & Vehicles
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Books"
          value="Books"
        >
          Books
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Business"
          value="Business"
        >
          Business
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Dating"
          value="Dating"
        >
          Dating
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Education"
          value="Education"
        >
          Education
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Entertainment"
          value="Entertainment"
        >
          Entertainment
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Events"
          value="Events"
        >
          Events
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Finance"
          value="Finance"
        >
          Finance
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Food & Drink"
          value="Food & Drink"
        >
          Food & Drink
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Health & Fitness"
          value="Health & Fitness"
        >
          Health & Fitness
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="House & Home"
          value="House & Home"
        >
          House & Home
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Music"
          value="Music"
        >
          Music
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Photography"
          value="Photography"
        >
          Photography
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Shopping"
          value="Shopping"
        >
          Shopping
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Social"
          value="Social"
        >
          Social
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Sports"
          value="Sports"
        >
          Sports
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Travel & Local"
          value="Travel & Local"
        >
          Travel & Local
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Adventure"
          value="Adventure"
        >
          Adventure
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="Other"
          value="Other"
        >
          Other
        </WithStyles(MenuItem)>
      </TextField>
      <LocationField
        bucketlist={
          Object {
            "category": "",
            "description": "",
            "dueDate": "",
            "location": "",
            "name": "",
            "privacy": "everyone",
          }
        }
        onChange={[Function]}
      />
      <PhotoField
        bucketlist={
          Object {
            "category": "",
            "description": "",
            "dueDate": "",
            "location": "",
            "name": "",
            "privacy": "everyone",
          }
        }
        changePhoto={[Function]}
        imageUrl=""
        removePhoto={[Function]}
      />
      <TextField
        label="privacy"
        name="privacy"
        onChange={[Function]}
        required={false}
        select={true}
        style={
          Object {
            "alignSelf": "center",
            "display": "flex",
            "justifyContent": "center",
            "width": "calc(100% - 40px)",
          }
        }
        value="everyone"
      >
        <WithStyles(MenuItem)
          key="everyone"
          value="everyone"
        >
          everyone
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="friends"
          value="friends"
        >
          friends
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          key="no one"
          value="no one"
        >
          no one
        </WithStyles(MenuItem)>
      </TextField>
    </div>
    <div
      className="actions"
    >
      <WithStyles(FlatButton)
        label="cancel"
        onClick={[Function]}
        style={
          Object {
            "color": "grey",
          }
        }
      />
      <WithStyles(FlatButton)
        disabled={true}
        keyboardFocused={true}
        label="save"
        onClick={[Function]}
      />
    </div>
  </div>
</WithStyles(Modal)>
`;
